From 5d76de0eee040fe5bbd5d99f9aca2ad4791747a4 Mon Sep 17 00:00:00 2001
From: kostap <kostap@marvell.com>
Date: Sun, 16 Nov 2014 15:22:50 +0200
Subject: [PATCH 0055/1345] fix: spi: a38x: Fix the value of size-cells
 parameter in DT

commit  6203e52b0a57e7fdd6b0abab45919165c3945eab from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

	- Correct the value of #size-cells SPI parameter in A38x
	  family boards DT.
	- The parameter must be "0" for SPI as defined in Kernel
	  documentation for SPI DT (spi-bus.txt)

Signed-off-by: kostap <kostap@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/14463
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Nadav Haklai <nadavh@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/14847

Conflicts:
	arch/arm/boot/dts/armada-385-db-ap.dts
	arch/arm/boot/dts/armada-385-db-gp.dts
	arch/arm/boot/dts/armada-385-db.dts
	arch/arm/boot/dts/armada-385-rd.dts
	arch/arm/boot/dts/armada-388-db-gp.dts
	arch/arm/boot/dts/armada-388-rd.dts

Make the change also in new armada-38* dts's

Change-Id: I1942a820560b49a1fce6a105ff6a7a217b75cfb2
Signed-off-by: Grzegorz Jaszczyk <jaz@semihalf.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/27577
Reviewed-by: Kostya Porotchkin <kostap@marvell.com>
Reviewed-by: Nadav Haklai <nadavh@marvell.com>
Tested-by: Lior Amsalem <alior@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/armada-380-gp.dts    |    2 +-
 arch/arm/boot/dts/armada-385-db-ap.dts |    2 +-
 arch/arm/boot/dts/armada-388-db.dts    |    2 +-
 arch/arm/boot/dts/armada-388-gp.dts    |    2 +-
 arch/arm/boot/dts/armada-388-rd.dts    |    2 +-
 5 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/arch/arm/boot/dts/armada-380-gp.dts b/arch/arm/boot/dts/armada-380-gp.dts
index 66b78dc..47774f7 100644
--- a/arch/arm/boot/dts/armada-380-gp.dts
+++ b/arch/arm/boot/dts/armada-380-gp.dts
@@ -69,7 +69,7 @@
 
 				spi-flash@0 {
 					#address-cells = <1>;
-					#size-cells = <1>;
+					#size-cells = <0>;
 					compatible = "st,m25p128";
 					reg = <0>; /* Chip select 0 */
 					spi-max-frequency = <50000000>;
diff --git a/arch/arm/boot/dts/armada-385-db-ap.dts b/arch/arm/boot/dts/armada-385-db-ap.dts
index 80c325d..291da24 100644
--- a/arch/arm/boot/dts/armada-385-db-ap.dts
+++ b/arch/arm/boot/dts/armada-385-db-ap.dts
@@ -266,7 +266,7 @@
 
 	spi-flash@0 {
 		#address-cells = <1>;
-		#size-cells = <1>;
+		#size-cells = <0>;
 		compatible = "st,m25p128", "jedec,spi-nor";
 		reg = <0>; /* Chip select 0 */
 		spi-max-frequency = <54000000>;
diff --git a/arch/arm/boot/dts/armada-388-db.dts b/arch/arm/boot/dts/armada-388-db.dts
index a727131..013f4dc 100644
--- a/arch/arm/boot/dts/armada-388-db.dts
+++ b/arch/arm/boot/dts/armada-388-db.dts
@@ -270,7 +270,7 @@
 
 	spi-flash@0 {
 		#address-cells = <1>;
-		#size-cells = <1>;
+		#size-cells = <0>;
 		compatible = "w25q32", "jedec,spi-nor";
 		reg = <0>; /* Chip select 0 */
 		spi-max-frequency = <108000000>;
diff --git a/arch/arm/boot/dts/armada-388-gp.dts b/arch/arm/boot/dts/armada-388-gp.dts
index 4b127b8..670c0e2 100644
--- a/arch/arm/boot/dts/armada-388-gp.dts
+++ b/arch/arm/boot/dts/armada-388-gp.dts
@@ -442,7 +442,7 @@
 
 	spi-flash@0 {
 		#address-cells = <1>;
-		#size-cells = <1>;
+		#size-cells = <0>;
 		compatible = "st,m25p128", "jedec,spi-nor";
 		reg = <0>; /* Chip select 0 */
 		spi-max-frequency = <50000000>;
diff --git a/arch/arm/boot/dts/armada-388-rd.dts b/arch/arm/boot/dts/armada-388-rd.dts
index af82f27..ddb1649 100644
--- a/arch/arm/boot/dts/armada-388-rd.dts
+++ b/arch/arm/boot/dts/armada-388-rd.dts
@@ -136,7 +136,7 @@
 
 	spi-flash@0 {
 		#address-cells = <1>;
-		#size-cells = <1>;
+		#size-cells = <0>;
 		compatible = "st,m25p128", "jedec,spi-nor";
 		reg = <0>; /* Chip select 0 */
 		spi-max-frequency = <108000000>;
-- 
1.7.9.5

